spec/index.bs

```diff
@@ -726,8 +720,62 @@ interface GPUBufferUsage {
 };
 </script>
 
+There are no restrictions on combining buffer usage flags, but some combinations
+man incur additional overhead. (For example, MAP_READ and STORAGE may incur a
+higher than expected overhead of copies, as the implemention must keep MAP_READ
+buffer mapping data as fresh as Fences require)
+
 ## Buffer Mapping ## {#buffer-mapping}
 
+<script type=idl>
+typedef unsigned long GPUMapFlags;
+interface GPUMap {
+    const GPUMapFlags READ  = 0x0001;
+    const GPUMapFlags WRITE = 0x0002;
+};
+</script>
+
+## {{GPUBuffer/map()}} ##
+
+If the devices is lost, returns null.
+
+If {{GPUBuffer/[[state]]}} is "executing", if {{GPUBuffer/[[readyAfterFenceID]]}}
+is greater than {{GPUQueue/[[readyFenceID]]}}, generate an error and return null.
+
+Otherwise, {{GPUBuffer/[[state]]}} becomes "ready".
+
+If {{GPUBuffer/[[state]]}} is not "ready", returns null.
+(Only one mapping may be active per buffer at a time)
+
+If {{GPUBuffer/map()}} includes {{GPUMap/READ}} but the buffer is not marked for
+{{GPUBufferUsage/MAP_READ}}, generate an error and return null.
+If {{GPUBuffer/map()}} includes {{GPUMap/WRITE}} but the buffer is not marked for
+{{GPUBufferUsage/MAP_WRITE}}, generate an error and return null.
+
+When buffers are used in a {{GPUCommandBuffer}} sent successfully to
+{{GPUQueue.submit()}}, {{[[state]]}} becomes "executing", and {{[[readyAfterFenceID]]}}
+becomes {{GPUQueue/[[nextFenceID]]}}.
+
+On {{GPUQueue/createFence()}} and {{GPUQueue/signal()}}:
+* In {{GPUFence/[[pendingfenceIDBySignalValue]]}}, assign key signalValue to value
+  {{GPUQueue/[[nextFenceID]]}}.
```

Comment:
Perhaps would be worth adding some wording somewhere about how you expect this to work with multi-queue.